### PR TITLE
Provide a script to generate code coverage

### PR DESCRIPTION
**Pull request by actione**
### Description

We previously implemented diff branch code coverage Per-PR in PR #576, but due to the installation limitations of code-cov, we temporarily abandoned this plan.
A script is provided here to facilitate generating code coverage locally and on automated systems. Later I will enable this to the internal automation system so that code coverage can be generated regularly

### CODE REVIEW

File: scripts/generate_cc.sh [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+
+# Usage:
+# bash scripts/generate_cc.sh -c
+# -or-
+# bash scripts/build_cudaq.sh -p
+# -or-
+# bash scripts/build_cudaq.sh -c -p
```
Comment (bmhowe23) [inline scripts/generate_cc.sh:16]: Perhaps a copy/paste error?
```suggestion
# bash scripts/generate_cc.sh -p
# -or-
# bash scripts/generate_cc.sh -c -p
```

**Review by bmhowe23 — COMMENTED** [review]
This will be nice to have ... thanks @actione.

File: scripts/generate_cc.sh (hunk 1 above)
Comment (actione) [reply scripts/generate_cc.sh:16]: Thanks, adding the -c and -p options at the same time means generating both C code coverage and python code coverage.

**Comment by actione** [comment]
Hi, @bmhowe23 , I think the arm64 caching error should be unrelated with the modification of this PR. We only added two llvm-cov components and their sizes are very small. I rebase this branch. Could you help retrigger CICD? thanks

File: scripts/generate_cc.sh (hunk 1 above)
Comment (bmhowe23) [reply scripts/generate_cc.sh:16]: > Thanks, adding the -c and -p options at the same time means generating both C code coverage and python code coverage.

I was meaning that I wonder if the references to `build_cudaq.sh` are correct. Shouldn't they be referencing `generate_cc.sh`?

Comment (actione) [reply scripts/generate_cc.sh]: Sorry, I didn't catch your meaning correctly :)

**Comment by actione** [comment]
Hi, @bmhowe23 , If there are no further actions or concerns, could you please help merge this PR so that I can generate code coverage regularly on the internal automation platform ASAP :).

**Pull request merged**